### PR TITLE
ci: Added ci-debugging.yml to enable local debugging

### PR DESCRIPTION
**Pull request by sarojsarab**
## Description
- Added ci-debugging.yml to enable local debugging
- Modularised docker image building with build-docker-image.yml

## Type of change
- ci-debugging.yml
- build-docker-image.yml

## How Has This Been Tested?
- Manual

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by Aishwarya-U-R** [comment]
@sarojsarab - Lets address the pointers we discussed on Slack & then proceed in this PR.

**Comment by sarojsarab** [comment]
> 

@Aishwarya-U-R , Added build-docker-image.yml to modularise the docker image building as we discussed.

File: .github/workflows/ci-debugging.yml [hunk 1]
```diff
@@ -0,0 +1,150 @@
+name: CI Debugging
+
+on:
+  # This line enables manual triggering of this workflow.
+  workflow_dispatch:
+
+jobs:
+  server-build:
+    name: server-build
+    uses: ./.github/workflows/server-build.yml
```
Comment (Aishwarya-U-R) [inline .github/workflows/ci-debugging.yml:10]: This means - server-build.yml is used again to build the server right?

Comment (sarojsarab) [reply .github/workflows/ci-debugging.yml:10]: Yeah we have to build server, client and rts once as we need this builds to build the docker image

Comment (Aishwarya-U-R) [reply .github/workflows/ci-debugging.yml:10]: Not built here again, but using the server already built from build-server stage.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**